android/modules/calendar/src/java/ti/modules/titanium/calendar/RecurrenceRuleProxy.java

```diff
@@ -0,0 +1,360 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2017 by Axway, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+package ti.modules.titanium.calendar;
+
+import org.appcelerator.kroll.KrollDict;
+import org.appcelerator.kroll.KrollProxy;
+import org.appcelerator.kroll.annotations.Kroll;
+import org.appcelerator.kroll.annotations.generator.JSONUtils;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.util.TiConvert;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Kroll.proxy
+public class RecurrenceRuleProxy extends KrollProxy {
+
+	// Keys for daysOfTheWeek dictionary.
+	private final String dayOfWeekKey = "daysOfWeek";
+	private final String weekNumberKey = "week";
+
+	// Keys for end dictionary.
+	private final String until = "endDate";
+	private final String count = "occurrenceCount";
+
+	// Field with the Date of the event begin to provide parity with iOS.
+	// Some information exposed can't be get solely from the recurrence rule.
+	private Date eventBegin;
+	private String rRule;
+
+	//region values that can be set from a Creation Dictionary
+	private int frequency = -1;
+	private int interval = -1;
+	private int[] daysOfTheMonth;
```

Comment:
I'm curious why the choice to use `int[]` for these fields that all seem to be either empty to have one single value. Personally I'd have used `Integer` type, with `null` representing a "no value".

Reply:
Can we use something like `Optional`: https://developer.android.com/reference/java/util/Optional.html or not until Api level 24 is our minimum?

Reply:
Our JavaScript API defines it as an array (link below). So, using an `int[]` is fine.
http://docs.appcelerator.com/platform/latest/#!/api/Titanium.Calendar.RecurrenceRule-property-daysOfTheMonth

@ypbnv, we need to double check what iOS returns if this property is not set upon creation. That is, if it returns null or an empty array. We don't document this and these details matter. Once known, we can then initialize the `int[]` member variables appropriately (either null or empty).

Reply:
@jquick-axway I agree. I used this as a reference:
https://developer.apple.com/documentation/eventkit/ekrecurrencerule/1507410-daysofthemonth?language=objc
Couldn't find the result there. Maybe try it on a device?

Reply:
@ypbnv, I just tested it out and confirmed that the following properties never return null on iOS. They return an empty array if not assigned upon creation. Personally, I agree with this behavior, because I find most script developers (unlike C/C++ devs) tend to not bother doing null checks.
- daysOfTheMonth
- daysOfTheWeek
- daysOfTheYear
- monthsOfTheYear
- weeksOfTheYear

So, we'll need to match the above behavior on Android.

Also, when you change the docs, would you mind updating the above properties with this info too please? Perhaps add the following sentence to the end of each property description like this...
`Will be an empty array if no recurrence of this type has been configured.`

Note that the `daysOfTheWeek` property is a dictionary and not an array. It definitely returns an empty dictionary for me on iOS in this case as well. So, update the docs appropriately.

Any case, I hope this helps!

Reply:
They are currently working this way on Android. I will add the description in the documentation.